pytest_factoryboy/fixture.py

```diff
@@ -1,51 +1,103 @@
+from functools import lru_cache
+from inspect import getmodule, signature
 
 import factory
 import factory.builder
 import factory.declarations
 import factory.enums
 import inflection
-import pytest
-
-from inspect import getmodule, signature
+import mako.template
 
 from pytest_factoryboy.compat import PostGenerationContext
 
+
 SEPARATOR = "__"
 
 
-FIXTURE_FUNC_FORMAT = """
-def {name}({deps}):
-    return _fixture_impl(request, **kwargs)
+module_template = mako.template.Template(
+    """\
+import pytest
+from pytest_factoryboy.fixture import (
+    attr_fixture,
+    factory_fixture,
+    model_fixture,
+    subfactory_fixture,
+)
+
+__all__ = (
+% for fixture_def in fixture_defs:
+    ${ repr(fixture_def.name)},
+% endfor
+)
+
+
+def _fixture(related):
+    def fixture_maker(fn):
+        fn._factoryboy_related = related
+        return pytest.fixture(fn)
+
+    return fixture_maker
+% for fixture_def in fixture_defs:
+
+
+${ fixture_def.kwargs_var_name } = {}
+
+
+@_fixture(related=${ repr(fixture_def.related) })
+def ${ fixture_def.name }(
+% for dep in ["request"] + fixture_def.deps:
+    ${ dep },
+% endfor
+):
+    return ${ fixture_def.function_name }(request, **${ fixture_def.kwargs_var_name })
+% endfor
 """
+)
 
 
-def make_fixture(name, module, func, args=None, related=None, **kwargs):
-    """Make fixture function and inject arguments.
+@dataclass
+class FixtureDef:
+    name: str
+    function_name: typing.Literal["model_fixture", "attr_fixture", "factory_fixture", "subfactory_fixture"]
+    function_kwargs: dict = field(default_factory=dict)
+    deps: list[str] = field(default_factory=list)
+    related: list[str] = field(default_factory=list)
 
-    :param name: Fixture name.
-    :param module: Python module to contribute the fixture into.
-    :param func: Fixture implementation function.
-    :param args: Argument names.
-    """
-    args = [] if args is None else list(args)
-    if "request" not in args:
-        args.insert(0, "request")
-    deps = ", ".join(args)
-    context = dict(_fixture_impl=func, kwargs=kwargs)
-    context.update(kwargs)
-    exec(FIXTURE_FUNC_FORMAT.format(name=name, deps=deps), context)
-    fixture_func = context[name]
-    fixture_func.__module__ = module.__name__
+    @property
+    def kwargs_var_name(self):
+        return f"_{self.name}__kwargs"
 
-    if related:
-        fixture_func._factoryboy_related = related
 
-    fixture = pytest.fixture(fixture_func)
-    setattr(module, name, fixture)
-    return fixture
+@lru_cache()  # This way we reuse the same folder for the whole execution of the program
+def make_temp_folder(package_name):
+    """Create a temporary folder and automatically delete it when the process exit."""
+    path = pathlib.Path(tempfile.mkdtemp()) / package_name
+    path.mkdir(parents=True, exist_ok=True)
+
+    atexit.register(shutil.rmtree, str(path))
+
+    return path
+
+
+def make_module(code: str, module_name: str, package_name: str):
```

Comment:
NIP: let's add missing type hinting for return type:

```suggestion
def make_module(code: str, module_name: str, package_name: str) -> ModuleType:
```